customer-commons/src/test/java/no/unit/nva/customer/service/impl/DynamoDBCustomerServiceTest.java

```diff
@@ -0,0 +1,230 @@
+        service.createCustomer(getNewCustomer());
+        service.createCustomer(getNewCustomer());
+
+        List<Customer> customers = service.getCustomers();
+        assertEquals(3, customers.size());
+    }
+
+    @Test
+    public void getCustomerNotFoundThrowsException() {
+        UUID nonExistingCustomer = UUID.randomUUID();
+        assertThrows(NotFoundException.class, () -> service.getCustomer(nonExistingCustomer));
+    }
+
+    @Test
+    public void getCustomerTableErrorThrowsException() {
+        Table failingTable = mock(Table.class);
+        when(failingTable.getItem(anyString(),any())).thenThrow(RuntimeException.class);
+        DynamoDBCustomerService failingService = new DynamoDBCustomerService(
+                objectMapper,
+                failingTable
+        );
+        DynamoDBException exception = assertThrows(DynamoDBException.class,
+            () -> failingService.getCustomer(UUID.randomUUID()));
+        assertEquals(ERROR_READING_FROM_TABLE, exception.getMessage());
+    }
+
+    @Test
+    public void getCustomersTableErrorThrowsException() {
+        Table failingTable = mock(Table.class);
+        when(failingTable.scan()).thenThrow(RuntimeException.class);
+        DynamoDBCustomerService failingService = new DynamoDBCustomerService(
+                objectMapper,
+                failingTable
+        );
+        DynamoDBException exception = assertThrows(DynamoDBException.class,
+            () -> failingService.getCustomers());
+        assertEquals(ERROR_READING_FROM_TABLE, exception.getMessage());
+    }
+
+    @Test
+    public void createCustomerTableErrorThrowsException() {
+        Table failingTable = mock(Table.class);
+        when(failingTable.putItem(any(Item.class))).thenThrow(RuntimeException.class);
+        DynamoDBCustomerService failingService = new DynamoDBCustomerService(
+                objectMapper,
+                failingTable
+        );
+        DynamoDBException exception = assertThrows(DynamoDBException.class,
+            () -> failingService.createCustomer(getNewCustomer()));
+        assertEquals(ERROR_WRITING_ITEM_TO_TABLE, exception.getMessage());
+
+    }
+
+    @Test
+    public void updateCustomerTableErrorThrowsException() {
+        Table failingTable = mock(Table.class);
+        when(failingTable.putItem(any(Item.class))).thenThrow(RuntimeException.class);
+        DynamoDBCustomerService failingService = new DynamoDBCustomerService(
+                objectMapper,
+                failingTable
+        );
+        Customer customer = getNewCustomer();
+        customer.setIdentifier(UUID.randomUUID());
+        DynamoDBException exception = assertThrows(DynamoDBException.class,
+            () -> failingService.updateCustomer(customer.getIdentifier(), customer));
+        assertEquals(ERROR_WRITING_ITEM_TO_TABLE, exception.getMessage());
+    }
+
+    @Test
+    public void customerToItemThrowsException() throws JsonProcessingException {
```

Comment:
maybe nitpicking, but the test name means that the method always throws an Exception

Reply:
Yes, it always throws an exception if the customer is not found :)